### PR TITLE
Submit select 25 tests from across files

### PR DESCRIPTION
**Pull request by Aishwarya-Hariharan-Iyer**
Tests finalised

### CODE REVIEW

File: client/src/pages/admin/CreateCategory.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+describe("Create Category Component", () => {
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+
+    describe("When the component is rendered", () => {
+        
+        test("The different components are rendered with successfull fetching of categories", async () => {
```
Comment (eugenetangkj) [inline client/src/pages/admin/CreateCategory.test.js]: If this test is not supposed to pass, should we use `test.failing`?

Comment (eugenetangkj) [inline client/src/pages/admin/CreateCategory.test.js]: 'Successful` is spelled wrongly

Comment (eugenetangkj) [reply client/src/pages/admin/CreateCategory.test.js]: Or is it supposed to pass, but the comment for assert is incorrect?

**Review by eugenetangkj — CHANGES_REQUESTED** [review]
Thanks! Looks good, except:
- The comments I left


- If you look at the CI workflow, there still seems to be errors (scroll all the way to the bottom). Are they handled by your tests?

1. In `categoryController.test.js`, there is a reference error for line 29 at ` await categoryController(req, res);`.  Is it cause it must be `import { categoryController } from './categoryController.js';`?
2. In `categoryController.test.js`, line 78, there's this line `expect(res.status).toHaveBeenCalledWith(409);`. Should it have been `test.failing`?

File: client/src/pages/admin/CreateCategory.test.js (hunk 1 above)
Comment (Aishwarya-Hariharan-Iyer) [reply client/src/pages/admin/CreateCategory.test.js]: Assertion corrected! Tysm!

Comment (Aishwarya-Hariharan-Iyer) [reply client/src/pages/admin/CreateCategory.test.js]: Spelling corrected! Tysm!

**Comment by Aishwarya-Hariharan-Iyer** [comment]
Tysm for the feedback! Here is a summary of the responses
> The comments I left
Resolved

> In categoryController.test.js, there is a reference error for line 29 at  await categoryController(req, res);. Is it cause it must be import { categoryController } from './categoryController.js';?

Since I did not write that test, I have not made any edits to it lest it interfere with the original tester's design and do not want to impact git commit history (incase the original tester has chosen this as part of their test suite). I think it is because it should be catgeoryControlller (deliberate spelling mistake in original code) but test uses only 2 l's. Have put this as a failing test upon request of original tester, but request original tester (as expressed in group) to double check.


> In categoryController.test.js, line 78, there's this line expect(res.status).toHaveBeenCalledWith(409);. Should it have been test.failing?

Since I did not write that test, I have not made any edits to it lest it interfere with the original tester's design and do not want to impact git commit history (incase the original tester has chosen this as part of their test suite).

**Comment by Aishwarya-Hariharan-Iyer** [comment]
Migrated to 3 fresh PRs for clarity

**Pull request closed without merging**